backends/cortex_m/ops/op_quantized_linear.cpp

```diff
@@ -0,0 +1,369 @@
+    header_->weight_zp_data_offset =
+        calculate_offset(weight_zero_point.const_data_ptr<int32_t>());
+    header_->bias_data_offset = bias.has_value()
+        ? calculate_offset(bias.value().const_data_ptr<int32_t>())
+        : 0; // 0 = no bias
+
+    // Create filter dims for validation
+    cmsis_nn_dims filter_dims = {in_feat, 1, 1, out_feat};
+    validate_size(filter_dims);
+  }
+
+  cmsis_nn_context get_cmsis_ctx() const {
+    cmsis_nn_context ctx;
+    ctx.buf = cmsis_workspace_ptr_; // Start after fat header
+    ctx.size = get_cmsis_workspace_size();
+    return ctx;
+  }
+
+  bool is_kernel_sum_updated() const {
+    return header_->kernel_sum_state.updated;
+  }
+
+  // Helper to compute kernel sums if not already done
+  void compute_kernel_sums_if_needed() {
+    if (!header_->kernel_sum_state.updated) {
+      arm_vector_sum_s8(
+          reinterpret_cast<int32_t*>(cmsis_workspace_ptr_),
+          header_->in_features,
+          header_->out_features,
+          get_weight_data(),
+          get_weight_zp_data()[0], // lhs_offset (weight zero point)
+          0, // rhs_offset
+          get_bias_data());
+      header_->kernel_sum_state.updated = true;
+      ET_LOG(
+          Info,
+          "Computed kernel sums. [required_bytes : %d]",
+          header_->kernel_sum_state.required_size);
+    }
+  }
+
+  // Getters that convert offsets back to pointers
+  const int8_t* get_weight_data() const {
+    return reinterpret_cast<const int8_t*>(
+        base_ptr_ + header_->weight_data_offset);
+  }
+
+  const int32_t* get_weight_zp_data() const {
+    return reinterpret_cast<const int32_t*>(
+        base_ptr_ + header_->weight_zp_data_offset);
+  }
+
+  const int32_t* get_bias_data() const {
+    return header_->bias_data_offset == 0
+        ? nullptr
+        : reinterpret_cast<const int32_t*>(
+              base_ptr_ + header_->bias_data_offset);
+  }
+
+  bool is_per_channel_quant() const {
+    return header_->is_per_channel;
+  }
+  int32_t get_in_features() const {
+    return header_->in_features;
+  }
+  int32_t get_out_features() const {
+    return header_->out_features;
+  }
+
+ private:
+  static constexpr size_t CMSIS_FAT_HEADER_SIZE = 32;
+
+  // Fixed-size metadata header using offsets for architecture independence.
+  // Stores cached weight information and kernel sum computation state.
+  // Ensures 32-byte alignment for the CMSIS workspace that follows.
+  struct CMSISFatHeader {
+    // Kernel sum computation state (8 bytes)
+    struct {
+      bool updated = false;
+      int32_t required_size = 0;
+    } kernel_sum_state;
+
+    // Cached weight tensor information
+    int32_t in_features;
+    int32_t out_features;
+
+    // Store offsets instead of raw pointers
+    uint32_t weight_data_offset;
+    uint32_t weight_zp_data_offset;
+    uint32_t bias_data_offset; // (0 if no bias)
+
+    bool is_per_channel;
+    uint8_t padding[3]; // 3 bytes explicit padding for 32-byte alignment
```

Comment:
You don't need any padding here, as long as this assert is OK,

`cmsis_workspace_ptr_ % 32 == 0`